src/service/registry/handler/base.go

```diff
@@ -0,0 +1,36 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"net/http"
+
+	"github.com/goharbor/harbor/src/pkg/reverseproxy"
+)
+
+// NewBaseHandler returns the handler to handle "/v2"
+func NewBaseHandler(proxy reverseproxy.ReverseProxy) http.Handler {
```

Comment:
isn't proxy an http.Handler?

Reply:
It is, we define a new type `ReverseProxy` to specify it must be a reverse proxy